File: 2017/04/high_entropy.go
package main

import (
	"fmt"
	"io/ioutil"
	"sort"
	"strings"
)

func readFile(filename string) (words [][]string) {
	dat, _ := ioutil.ReadFile("input")
	lines := strings.Split(string(dat), "\n")
	for _, line := range lines {
		words = append(words, strings.Split(line, " "))
	}
	return
}

func main() {
	fileContent := readFile("input")
	count := 0

	for _, line := range fileContent {
		words := make(map[string]int)
		duplicates := false
		for _, word := range line {
			_, ok := words[word]
			if ok {
				duplicates = true
				break
			}
			words[word] = 1
		}
		if !duplicates {
			count++
		}
	}

	fmt.Println(count)

	count = 0
	for _, line := range fileContent {
		words := make(map[string]int)
		duplicates := false
		for _, word := range line {
			w := strings.Split(word, "")
			sort.Strings(w)
			word = strings.Join(w, "")
			//fmt.Println(word)
			_, ok := words[word]
			if ok {
				duplicates = true
				break
			}
			words[word] = 1
		}
		if !duplicates {
			count++
		}
	}

	fmt.Println(count)

}

File: 2017/04/high_entropy.py
#env python3

with open("input") as f:
    lines = f.readlines()


count = 0

for line in lines:
    words = dict()
    for word in line.strip().split(" "):
        if word in words:
            break
        words[word] = 1
    else:
        count += 1
    
print(count)


count = 0

for line in lines:
    words = dict()
    for word in line.strip().split(" "):
        word = "".join(sorted(list(word)))
        if word in words:
            break
        words[word] = 1
    else:
        count += 1
    
print(count)